src/rgw/rgw_rest_swift.cc

```diff
@@ -1846,3 +1992,11 @@ RGWHandler_REST* RGWRESTMgr_SWIFT::get_handler(struct req_state *s)
 
   return new RGWHandler_REST_Obj_SWIFT;
 }
+
+RGWHandler_REST* RGWRESTMgr_SWIFT_Info::get_handler(struct req_state *s)
+{
+  if (!RGWRESTMgr_SWIFT::get_handler(s))
+    return nullptr;
```

Comment:
this would leak the memory allocated by `RGWRESTMgr_SWIFT::get_handler()`. you probably just want to call `RGWHandler_REST_SWIFT::init_from_header()` here instead